vyper/compiler/input_bundle.py

```diff
@@ -1,19 +1,22 @@
 import contextlib
 import json
 import os
-from dataclasses import dataclass
+from dataclasses import asdict, dataclass, field
 from functools import cached_property
 from pathlib import Path, PurePath
-from typing import Any, Iterator, Optional
+from typing import TYPE_CHECKING, Any, Iterator, Optional
 
 from vyper.exceptions import JSONError
 from vyper.utils import sha256sum
 
 # a type to make mypy happy
 PathLike = Path | PurePath
 
+if TYPE_CHECKING:
```

Comment:
This construction is common for circular references, it's not clear why it's used here. Performance maybe?

Reply:
it's to clarify that the import is not really necessary except to type check. we don't actually use functions or classes from `zipfile`